CRM/MembershipExtras/Form/MembershipPeriod/Activate.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+use CRM_MembershipExtras_ExtensionUtil as E;
+
+/**
+ * Membership period activate form controller class.
+ */
+class CRM_MembershipExtras_Form_MembershipPeriod_Activate extends CRM_MembershipExtras_Form_MembershipPeriod_Base {
+
+  /**
+   * @inheritdoc
+   */
+  protected function setFormTitle() {
+    CRM_Utils_System::setTitle(E::ts('Activate Membership Period?'));
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public function postProcess() {
+    $period = new CRM_MembershipExtras_BAO_MembershipPeriod();
```

Comment:
you can use the BAO create() method to update the period instead, same in CRM_MembershipExtras_Form_MembershipPeriod_Deactivate.

Reply:
Tried to use create, but since we now need to update membership dates too, I pulled in the stuff you did for RSE-78 and used the updatePeriod method.